app/src/components/token/Token.tsx

```diff
@@ -0,0 +1,252 @@
+  padding: 0 var(--ac-global-dimension-static-size-100);
+  border-radius: var(--ac-global-rounding-large);
+  border: 1px solid
+    lch(from var(--ac-internal-token-color) calc((l) * infinity) c h / 0.3);
+  color: lch(from var(--ac-internal-token-color) calc((50 - l) * infinity) 0 0);
+  user-select: none;
+
+  &[data-size="S"] {
+    height: var(--ac-global-dimension-static-size-200);
+  }
+
+  &[data-size="M"] {
+    height: var(--ac-global-dimension-static-size-250);
+  }
+
+  &[data-size="L"] {
+    height: var(--ac-global-dimension-static-size-300);
+  }
+
+  &[data-disabled] {
+    opacity: 0.5;
+    cursor: not-allowed;
+  }
+
+  &[data-theme="light"] {
+    background: var(--ac-internal-token-color);
+  }
+
+  &[data-theme="dark"] {
+    // generate a new dark token bg color from the input color
+    --px-internal-dark-token-bg: lch(
+      from var(--ac-internal-token-color) l c h / calc(alpha - 0.8)
+    );
+    background: var(--px-internal-dark-token-bg);
+    // generate a new dark token text color from the input color
+    color: lch(
+      from var(--px-internal-dark-token-bg) calc((l) * infinity) c h / 1
+    );
+  }
+
+  &[data-interactive]:not([data-disabled]) {
+    cursor: pointer;
+
+    > button {
+      &:focus-visible {
+        outline: 2px solid var(--ac-focus-ring-color);
+        border-radius: var(--ac-global-rounding-small);
+      }
+    }
+  }
+
+  &[data-removable] {
+    padding-right: var(--ac-global-dimension-static-size-25);
+  }
+
+  > button {
+    all: unset;
+    cursor: pointer;
+    display: flex;
+    align-items: center;
+
+    &[disabled] {
+      cursor: not-allowed;
+    }
+  }
+`;
+
+function TokenLeadingVisual({
+  children,
+  size = "M",
+}: React.PropsWithChildren<SizingProps>) {
+  return (
+    <span
+      data-size={size}
+      css={css`
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        width: var(--ac-global-dimension-static-size-200);
+        height: var(--ac-global-dimension-static-size-200);
+
+        &[data-size="M"] {
+          margin-right: var(--ac-global-dimension-static-size-50);
+        }
+
+        &[data-size="L"] {
+          margin-right: var(--ac-global-dimension-static-size-100);
+        }
+      `}
+    >
+      {children}
+    </span>
+  );
+}
+
+/**
+ * A token is a pill or tag-like component that can display a string of text with optional interactions.
+ * It can take one of four forms:
+ * 1. Default: Wrapped child content
+ * 2. Interactive: Wrapped button with child content that can be clicked (onPress)
+ * 3. Removable: Wrapped child content, sibling to a remove button (onRemove)
+ * 4. Full Interactive: Wrapped sibling buttons with child content in the non-remove button
+ */
+function Token({
+  children,
+  isDisabled,
+  css: cssProp,
+  color = "var(--ac-global-color-grey-300)",
+  onPress,
+  onRemove,
+  size = "M",
+  style,
+  leadingVisual,
+  ...rest
+}: TokenProps): JSX.Element {
```

Comment:
```suggestion
}: TokenProps) {
```

nit: can't this just be inferred?

Reply:
Ah yeah this is a holdover from a much worse design I had originally with function overloads 😂

Reply:
Removed